File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;
class Solution {
    public String[] solution(String[] players, String[] callings) {
        HashMap<String, Integer> player = new HashMap<>();
        
        for(int i = 0; i<players.length; i++) player.put(players[i], i);
        
        for(String call : callings){
            int idx = player.get(call);
            
            String temp = players[idx - 1];
            players[idx - 1] = players[idx];
            players[idx] = temp;
            
            player.put(players[idx - 1], idx - 1);
            player.put(players[idx], idx);
        }
        
        return players;
    }
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    let map = new Map()
    
    for(var player of players) map.set(player, players.indexOf(player))
    
    for(var i = 0; i<callings.length; i++){
        var idx = map.get(callings[i])
        
        var temp = players[idx]
        players[idx] = players[idx - 1]
        players[idx - 1] = temp
        
        map.set(players[idx - 1], idx - 1)
        map.set(players[idx], idx)
    }
    return players;
}